controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -118,6 +119,26 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (controller.OperationRe
 		return controller.ContinueProcessing()
 	}
 
+	// The default value for ErrorOccured is 'True'.  We know that the state
+	// of the condition is still 'True' due to the earlier call to
+	// HaveBindingsFailed().  If this condition is still true after a
+	// reasonable time then we assume that the SEB is stuck in an unrecoverable
+	// state and clean it up.  Otherwise we requeue and wait until the timeout
+	// has passed
+	const timeoutMinutes float64 = 5
+	var lastTransitionTime time.Time
+	for _, condition := range a.snapshotEnvironmentBinding.Status.BindingConditions {
+		if condition.Type == gitops.BindingErrorOccurredStatusConditionType {
+			lastTransitionTime = condition.LastTransitionTime.Time
+		}
+	}
+	if time.Since(lastTransitionTime).Minutes() < timeoutMinutes {
+		a.logger.Info("The SnapshotEnvironmentBinding is younger than the threshold; requeuing cleanup after delay")
+		return controller.RequeueAfter(time.Duration(timeoutMinutes*float64(time.Minute)), nil)
+	} else {
+		a.logger.Info(fmt.Sprintf("SEB is old. Age: %s, time.Now: %s", lastTransitionTime, time.Now()))
```

Comment:
Similar to above, rephrasing tihs to something like `SEB has been in the Error state for more than the threshold time of %s`.